examples/cim/gnn/simple_gnn.py

```diff
@@ -0,0 +1,335 @@
+            self.act_1 = GeLU()
+            self.fc_2 = nn.Linear(input_size, output_size)
+        elif net_type == "2layer":
+            self.fc_0 = nn.Linear(input_size, hidden_size)
+            self.act_0 = GeLU()
+            # self.do_0 = Dropout(dropout)
+            self.fc_1 = nn.Linear(hidden_size, hidden_size // 2)
+            self.act_1 = GeLU()
+            self.fc_2 = nn.Linear(hidden_size // 2, output_size)
+        elif net_type == "1layer":
+            self.fc_0 = nn.Linear(input_size, hidden_size)
+            self.act_0 = GeLU()
+            self.fc_1 = nn.Linear(hidden_size, output_size)
+
+    def forward(self, x):
+        if self.net_type == "res":
+            x1 = self.act_0(self.fc_0(x))
+            x1 = self.act_1(self.fc_1(x1) + x)
+            return self.fc_2(x1)
+        elif self.net_type == "2layer":
+            x = self.act_0(self.fc_0(x))
+            x = self.act_1(self.fc_1(x))
+            x = self.fc_1(x)
+            return x
+        else:
+            x = self.fc_1(self.act_0(self.fc_0(x)))
+            return x
+
+
+class SharedAC(nn.Module):
+    """The actor-critic module shared with multiple agents.
+
+    This module maps the input graph of the observation to the policy and value space. It first extracts the temporal
+    information separately for each node with a small transformer block and then extracts the spatial information with
+    a multi-graph/channel graph attention. Finally, the extracted feature embedding is fed to a actor header as well
+    as a critic layer, which are the two MLPs with residual connections.
+    """
+
+    def __init__(
+            self, input_dim_p, edge_dim_p, input_dim_v, edge_dim_v, tick_buffer, action_dim, a=True, c=True,
+            scale=4, ac_head="res"):
+        super().__init__()
+        assert(a or c)
+        self.a, self.c = a, c
+        self.input_dim_v = input_dim_v
+        self.input_dim_p = input_dim_p
+        self.tick_buffer = tick_buffer
+
+        self.pre_dim_v, self.pre_dim_p = 8 * scale, 16 * scale
+        self.p_pre_layer = nn.Sequential(
+            nn.Linear(input_dim_p, self.pre_dim_p), GeLU(), PositionalEncoder(
+                d_model=self.pre_dim_p, max_seq_len=tick_buffer))
+        self.v_pre_layer = nn.Sequential(
+            nn.Linear(input_dim_v, self.pre_dim_v), GeLU(), PositionalEncoder(
+                d_model=self.pre_dim_v, max_seq_len=tick_buffer))
+        p_encoder_layer = TransformerEncoderLayer(
+            d_model=self.pre_dim_p, nhead=4, activation="gelu", dim_feedforward=self.pre_dim_p * 4)
+        v_encoder_layer = TransformerEncoderLayer(
+            d_model=self.pre_dim_v, nhead=2, activation="gelu", dim_feedforward=self.pre_dim_v * 4)
+        # self.trans_layer_p = TransformerEncoder(p_encoder_layer, num_layers=3, norm=Norm(self.pre_dim_p))
+        # self.trans_layer_v = TransformerEncoder(v_encoder_layer, num_layers=3, norm=Norm(self.pre_dim_v))
```

Comment:
here

Reply:
this is usefull, to demonstrate alternative ways to initialize it